### PR TITLE
Update sandbox component 'opennlp-coref' to be compatible with latest opennlp-tools release

### PR DESCRIPTION
**Pull request by mawiesne**
- adjusts `opennlp-tools` to 2.1.0
- adjusts parent project (org.apache.apache) to version `18`
- adjusts Java language level to 11

### CODE REVIEW

File: opennlp-coref/src/main/java/opennlp/tools/cmdline/coref/CoreferencerTool.java [hunk 1]
```diff
@@ -85,8 +85,7 @@ private void show(Parse p) {
         System.out.print(" ");
       }
       Parse[] children = p.getChildren();
-      for (int pi = 0, pn = children.length; pi < pn;pi++) {
-        Parse c = children[pi];
+      for (Parse c : children) {
```
Comment (kinow) [inline opennlp-coref/src/main/java/opennlp/tools/cmdline/coref/CoreferencerTool.java:88]: :ok_man:

File: opennlp-coref/src/main/java/opennlp/tools/formats/CorefSampleStreamFactory.java [hunk 2]
```diff
@@ -42,16 +44,19 @@ public static void registerFactory() {
...
-
-    ObjectStream<String> lineStream = new ParagraphStream(new PlainTextByLineStream(sampleDataIn
-        .getChannel(), params.getEncoding()));
-
-    return new CorefSampleDataStream(lineStream);
+    try {
+      MarkableFileInputStreamFactory factory = new MarkableFileInputStreamFactory(params.getData());
+      ObjectStream<String> lineStream = new ParagraphStream(new PlainTextByLineStream(
+              factory, params.getEncoding()));
+      return new CorefSampleDataStream(lineStream);
```
Comment (kinow) [inline opennlp-coref/src/main/java/opennlp/tools/formats/CorefSampleStreamFactory.java:57]: Ditto what I said in the other PR, if it makes sense, these factory and stream could be in the `try-with-resources`, assuming they are closeables and it makes sense :+1:

File: opennlp-coref/pom.xml [hunk 3]
```diff
@@ -25,20 +25,20 @@
 	<parent>
 		<groupId>org.apache</groupId>
 		<artifactId>apache</artifactId>
-		<version>13</version>
+		<version>18</version>
```
Comment (rzo1) [inline opennlp-coref/pom.xml:28]: 29?

File: opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java [hunk 4]
```diff
@@ -169,7 +170,14 @@ public DiscourseEntity retain(MentionContext mention, DiscourseModel dm) {
       DiscourseEntity cde = dm.getEntity(ei);
       MentionContext cec = cde.getLastExtent(); // candidate extent context
       if (cec.getId() == mention.getId()) {
-        distances.add(ei);
+        // adding counts
+        Integer count = distances.get(ei);
```
Comment (rzo1) [inline opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java:174]: `distances.merge(e, 1, Integer::sum)` ?

File: opennlp-coref/src/main/java/opennlp/tools/coref/resolver/MaxentResolver.java [hunk 5]
```diff
@@ -286,7 +294,14 @@ public DiscourseEntity retain(MentionContext mention, DiscourseModel dm) {
               events.add(new Event(SAME, features.toArray(new String[features.size()])));
               de = cde;
               //System.err.println("MaxentResolver.retain: resolved at "+ei);
-              distances.add(ei);
+              // adding counts
+              Integer count = distances.get(ei);
```
Comment (rzo1) [inline opennlp-coref/src/main/java/opennlp/tools/coref/resolver/MaxentResolver.java:298]: `distances.merge(e, 1, Integer::sum)` ?

**Review by rzo1 — APPROVED** [review]
lgtm

File: opennlp-coref/pom.xml (hunk 3 above)
Comment (mawiesne) [reply opennlp-coref/pom.xml:28]: Not yet, future work. See other comment in other PR.

File: opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java (hunk 4 above)
Comment (mawiesne) [reply opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java:174]: Will check.

File: opennlp-coref/src/main/java/opennlp/tools/coref/resolver/MaxentResolver.java (hunk 5 above)
Comment (mawiesne) [reply opennlp-coref/src/main/java/opennlp/tools/coref/resolver/MaxentResolver.java:298]: Will check.

File: opennlp-coref/pom.xml (hunk 3 above)
Comment (rzo1) [reply opennlp-coref/pom.xml:28]: https://issues.apache.org/jira/browse/OPENNLP-1452

File: opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java (hunk 4 above)
Comment (mawiesne) [reply opennlp-coref/src/main/java/opennlp/tools/coref/resolver/AbstractResolver.java:174]: `e` -> `ei`

File: opennlp-coref/pom.xml (hunk 3 above)
Comment (mawiesne) [reply opennlp-coref/pom.xml:28]: I'll leave a TODO referencing this issue number here.

**Review by rzo1 — APPROVED** [review]

**Pull request merged**